crates/node/core/src/node_config.rs

```diff
@@ -70,8 +70,8 @@ use tracing::*;
 ///     let builder = builder.with_rpc(rpc);
 /// }
 /// ```
-#[derive(Debug, Clone)]
-pub struct NodeConfig {
+#[derive(Debug)]
+pub struct NodeConfig<ChainSpec> {
```

Comment:
~~this is a bit unpleasant, but can we impls Clone here manually, auto derive doesn't work for arc.~~

nvm saw it below

maybe we could also remove the Arc and require chainspec clone and expect that chainspec internally wraps the arc, but different discussion